scripts/generate/templates/TransientStorage.js

```diff
@@ -0,0 +1,47 @@
+const format = require('../format-lines');
+const { capitalize } = require('../../helpers');
+
+const TYPES = ['address', 'bool', 'bytes32', 'uint256'];
+
+const header = `\
+pragma solidity ^0.8.24;
+
+/**
+ * @dev Library for reading and writing transient storage.
+ *
+ * Note: This library only works on networks where EIP-1153 is available.
+ */
+`;
+
+const load = type => `\
+/**
+ * @dev Returns the transient slot \`slot\` as a \`${type}\`.
+ */
+function load${capitalize(type)}(bytes32 slot) internal view returns (${type} r) {
```

Comment:
Any reason in particular to make the return value different to `StorageSlot`? The values there are defined as:

```
 struct Uint256Slot {
    uint256 value;
}
```

I definitely think this syntax is better, but I'm wondering if we should prefer one syntax over the other.

Reply:
AFAIK, Solidity supports structs in memory and in storage, but not transient ones. You can't mark it. Only thing we have is low level opcode access through assembly ?

Reply:
![Capture d’écran du 2024-03-14 20-49-50](https://github.com/OpenZeppelin/openzeppelin-contracts/assets/2432299/37c34912-e0ee-4a34-a2f2-5dc2930e5969)

Reply:
![Capture d’écran du 2024-03-14 20-50-54](https://github.com/OpenZeppelin/openzeppelin-contracts/assets/2432299/7a448b2d-7ce2-460a-8fbb-85ac3c063c7b)

Reply:
See https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4955/files#r1525423381